Use rest_framework's exception_handler in DEBUG

Twixes opened the pull request:
This should expose `rest_framework`'s `exception_handler` function instead of `exceptions_hog`'s in DEBUG mode. Disableable with setting `EXCEPTIONS_HOG` subsetting `ENABLE_IN_DEBUG`.

paolodamico reviewed (changes requested):
A few thoughts,
- The way DRF's [exception_handler](https://github.com/encode/django-rest-framework/blob/91916a4db14cd6a06aca13fb9a46fc667f6c0682/rest_framework/views.py#L71) works so that Django raises exceptions normally is returning `None` when the exception does not inherit from `APIException`. I think we should do the same thing here when `DEBUG=1`, this way you'll still get the custom error responses for DRF-based exceptions (e.g. a validation error).
- We should add test cases for everything.
- We need to handle the case of settings not having a `DEBUG` attribute (test case here).
- Great idea adding the `ENABLE_IN_DEBUG` config, however we also need to register this setting in `settings.py` for it to work (we should add a test case here too, this would've actually failed).
- I would prefer to keep the line length at 89 for this project, we don't have complex queries that take a lot of space like the main repo and this way we can support better contributions from smaller screens, but definitely open to discuss this.
- Finally, we should document this functionality in the README so users know the default behavior and how to override it.

These are quite a few things, so let me know if you prefer I make the changes.

Twixes commented:
Sure, thanks for the feedback. Changed the approach to just return `None` from the handler. Added some unit tests, will also add API ones. `ENABLE_IN_DEBUG` is now correctly handled with a default of `False`. I find 120 chars the most practical, but restored 89 here, it's outside of this PR's scope anyway. TODO: documentation in README.

Twixes commented on tests/test_handler.py:
```diff
@@ -112,26 +119,70 @@ def test_protected_error() -> None:
+
+def test_not_found_exception_debug(settings, res_not_found) -> None:
+    settings.DEBUG = True
+    # Same as normal, since APIException instance
+    response = exception_handler(exceptions.NotFound())
+    assert response is not None
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+    assert response.data == res_not_found
+
+    # Test Django base exception too
+    # Not handled, since not APIException instance
+    # (Django errors only inherit from the Python Exception)
+    response = exception_handler(Http404())
+    assert response is None
```
This is tricky. Django errors don't inherit from a common Django base error, just from the Python `Exception`, and checking for each Django error would be cumbersome and brittle. Thoughts? @paolodamico

paolodamico replied:
Addressed

Twixes commented:
Unit tests all work as intended, can't get API tests to take changed `settings.DEBUG` value into account. 🤔
As for documentation, I suggest you add `drf-exceptions-hog` settings info @paolodamico, as it seems that `EXCEPTION_REPORTING` also needs documenting.
